torch/csrc/utils/torch_dispatch_mode.h

```diff
@@ -23,6 +23,34 @@ struct StashTorchDispatchModeGuard {
   std::shared_ptr<at::SafePyObject> saved_mode_;
 };
 
+struct ProxyOrFakeModeGuard {
+ public:
+  ProxyOrFakeModeGuard(bool is_proxy) {
```

Comment:
An enum seems like it would be much better here

Reply:
Actually, instead of having parallel APIs for `proxy_mode` and `fake_mode`, it might be easier to define a enum, similar to DispatchKey, which contains the special modes that we want to take care of here.

Reply:
agreed, will do